src/coreclr/vm/eventtrace.cpp

```diff
@@ -4566,9 +4566,13 @@ VOID ETW::MethodLog::SendMethodJitStartEvent(MethodDesc *pMethodDesc, SString *n
                 ulMethodToken = (ULONG)0;
         }
         else
+        {
             ulMethodToken = (ULONG)pMethodDesc->GetMemberDef();
+        }
 
-        if(pMethodDesc->IsIL())
+        // An IL method that has no IL header can occur for dynamically
+        // generated code during JIT (for example, UnsafeAccessor).
+        if(pMethodDesc->IsIL() && pMethodDesc->GetILHeader() != NULL)
```

Comment:
`GetILHeader` is not trivial operation. We should avoid calling it twice.

(It would be even better to let caller pass it in so that it matches what was actually JITed. `GetILHeader` does not provide stable results in the presence of rejit.)

Reply:
> It would be even better to let caller pass it in so that it matches what was actually JITed. GetILHeader does not provide stable results in the presence of rejit.

That is going to make this PR a bit larger. Is that what you want?

Reply:
I do not have a problem with making the PR a bit larger. I do not see a significant risk with doing the right thing here.

Reply:
I don't even think we should consider this. It is far too much change right now. I will add this to my backlog and handle it later.

Reply:
Didn't see your reply. Okay. I will give it a shot.

Reply:
(It is fine with me to do the cleanup later.)

Reply:
It was simple, I just wasn't sure how much churn we wanted. Let me know if you'd like me to revert it.